x-pack/plugins/apm/public/components/app/ServiceDetails/index.tsx

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { connect } from 'react-redux';
+// @ts-ignore
+import { getLicense } from 'x-pack/plugins/apm/public/store/reactReduxRequest/license';
+import { hasDynamicBaseline } from 'x-pack/plugins/apm/public/store/reactReduxRequest/transactionOverviewCharts';
+import { IReduxState } from 'x-pack/plugins/apm/public/store/rootReducer';
+import { getUrlParams } from 'x-pack/plugins/apm/public/store/urlParams';
+import { ServiceDetailsView } from './view';
+
+function mapStateToProps(state = {} as IReduxState) {
+  return {
+    urlParams: getUrlParams(state),
+    hasDynamicBaseline: hasDynamicBaseline(state),
+    license: getLicense(state)
+  };
+}
+
+const ServiceDetails = connect(mapStateToProps)(ServiceDetailsView);
+
+export { ServiceDetails };
```

Comment:
Heh, I actually hate it, but Chandler and I discovered through a ton of trial and error that this is the only way for the export to inherit the variable name as the "function.name" value, and then get pulled in as React's "displayName". In other words, doing it like this means it will show up as `Connect<ServiceDetails>` in snapshots but doing it like `export const ServiceDetails =` will mean it shows up as `Connect<Component>` :/

Reply:
Oooh, that's interesting!